yt_dlp/extractor/nekohacker.py

```diff
@@ -0,0 +1,223 @@
+                    'track': 'むじな de なじむ feat. 六科なじむ (CV: 日高里菜 )',
+                    'track_number': 2,
+                    'duration': None
+                }
+            },
+            {
+                'url': 'https://nekohacker.com/wp-content/uploads/2021/01/進め！むじなカンパニー-instrumental.mp3',
+                'md5': 'adde9e9a16e1da5e602b579c247d0fb9',
+                'info_dict': {
+                    'id': '710',
+                    'ext': 'mp3',
+                    'title': '進め！むじなカンパニー (instrumental)',
+                    'thumbnail': 'https://nekohacker.com/wp-content/uploads/2021/01/OP表-1024x1024.png',
+                    'vcodec': 'none',
+                    'acodec': 'mp3',
+                    'release_date': '20210115',
+                    'album': '進め！むじなカンパニー',
+                    'artist': 'Nekohacker',
+                    'track': '進め！むじなカンパニー (instrumental)',
+                    'track_number': 3,
+                    'duration': None
+                }
+            },
+            {
+                'url': 'https://nekohacker.com/wp-content/uploads/2021/01/むじな-de-なじむ-instrumental.mp3',
+                'md5': 'ebb0443039cf5f9ff7fd557ed9b23599',
+                'info_dict': {
+                    'id': '712',
+                    'ext': 'mp3',
+                    'title': 'むじな de なじむ (instrumental)',
+                    'thumbnail': 'https://nekohacker.com/wp-content/uploads/2021/01/OP表-1024x1024.png',
+                    'vcodec': 'none',
+                    'acodec': 'mp3',
+                    'release_date': '20210115',
+                    'album': '進め！むじなカンパニー',
+                    'artist': 'Nekohacker',
+                    'track': 'むじな de なじむ (instrumental)',
+                    'track_number': 4,
+                    'duration': None
+                }
+            }
+        ]
+    }]
+
+    def _real_extract(self, url):
+        playlist_id = self._match_id(url)
+
+        webpage = self._download_webpage(url, playlist_id)
+        playlist = get_element_by_class('playlist', webpage)
+
+        # handle cases which match _VALID_URL, but no element is found
+        if playlist is None:
+            try:
+                iframe = get_element_text_and_html_by_tag('iframe', webpage)
+            except ValueError:
+                iframe = None
+
+            # handle iframes instead of playlist
+            if iframe is not None:
+                attr = extract_attributes(iframe[1])
+                iframesrc = attr['src']
+                # handle spotify explicitly
+                if iframesrc is not None and re.match(r'https?://(?:www\.|open\.)?spotify\.com', iframesrc):
+                    raise ExtractorError('no playlist found, but iframe with spotify', expected=True)
+                raise ExtractorError('no playlist element found, but a unhandled iframe')
+
+            raise ExtractorError('no playlist element found - likely not a album', expected=True)
```

Comment:
- `utils.try_call` makes it easy when you're setting a var and need a simple try/except

 - I think we don't want to expect the first ExtractorError in my suggestion, since if neither a playlist or iframe is found in the page, that means the website's overall format could have changed and we'd want that to be reported.

 - If the iframe embed is not Spotify, we could try returning the webpage URL as a `url_result` to the generic extractor and seeing if it can detect any embeds that yt-dlp supports (if not it will raise UnsupportedError)
```suggestion
        if not playlist:
            iframe = try_call(lambda: get_element_text_and_html_by_tag('iframe', webpage)[1]) or ''
            iframe_src = url_or_none(extract_attributes(iframe).get('src'))
            if not iframe_src:
                raise ExtractorError('No playlist or embed found in webpage')
            elif re.match(r'https?://(?:\w+\.)?spotify\.com/', iframe_src):
                raise ExtractorError('Spotify embeds are not supported', expected=True)
            return self.url_result(url, 'Generic')
```